### PR TITLE
[Instagram] Try bypassing login wall with embed page

### LINKED ISSUE #2093: [Instragam] ERROR: You need to log in to access this content
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2021.12.01**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

France

### Description

Hi, 

This URL plays in my browser: https://www.instagram.com/p/CTomrHIFc36/ 

However, **it fails to download with yt-dlp** (see attached log). 

Nevertheless, I noticed that **after many retries**, even in my browser, **I get a login page instead of the video content**. 

I guess Instagram is applying a rate limit to the number of pages of content that can be seen without loging-in from a given IP address. Is there a way for yt-dlp to bypass this limitation? (using some API, different URL or else?)

_Note: I'm actually running the very last commit of yt-dlp, that is b28bac93a._ 

Thanks, 

### Verbose log

```shell
[debug] Command-line config: ['-v', 'https://www.instagram.com/p/CTomrHIFc36/']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, err UTF-8, pref UTF-8
[debug] yt-dlp version 2021.12.01 [91f071af6]
[debug] Python version 3.7.3 (CPython 64bit) - Linux-5.10.0-0.bpo.8-amd64-x86_64-with-debian-10.11
[debug] exe versions: ffmpeg 4.4.1-2 (fdk,setts), ffprobe 4.4.1-2
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [Instagram] Extracting URL: https://www.instagram.com/p/CTomrHIFc36/
[Instagram] CTomrHIFc36: Downloading webpage
ERROR: [Instagram] CTomrHIFc36: You need to log in to access this content. Use --cookies, --username and --password, or --netrc to provide account credentials
  File "/usr/lib/python3/dist-packages/yt_dlp/extractor/common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/yt_dlp/extractor/instagram.py", line 296, in _real_extract
    self.raise_login_required('You need to log in to access this content')
  File "/usr/lib/python3/dist-packages/yt_dlp/extractor/common.py", line 1089, in raise_login_required
    raise ExtractorError(msg, expected=True)

------------------

# AT SOME POINT, AFTER SEVERAL RETIRES, IT DID WORK, HERE IS THE *WORKING* LOG BELOW: 

[debug] Command-line config: ['-v', 'https://www.instagram.com/p/CTomrHIFc36/']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, err UTF-8, pref UTF-8
[debug] yt-dlp version 2021.12.01 [91f071af6]
[debug] Python version 3.7.3 (CPython 64bit) - Linux-5.10.0-0.bpo.8-amd64-x86_64-with-debian-10.11
[debug] exe versions: ffmpeg 4.4.1-2 (setts,fdk), ffprobe 4.4.1-2
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [Instagram] Extracting URL: https://www.instagram.com/p/CTomrHIFc36/
[Instagram] CTomrHIFc36: Downloading webpage
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] CTomrHIFc36: Downloading 1 format(s): 0
[debug] Invoking downloader on "https://scontent-cdt1-1.cdninstagram.com/v/t50.2886-16/241636096_184700850417257_4727285400846218921_n.mp4?efg=eyJ2ZW5jb2RlX3RhZyI6InZ0c192b2RfdXJsZ2VuLjcyMC5mZWVkLmRlZmF1bHQiLCJxZV9ncm91cHMiOiJbXCJpZ193ZWJfZGVsaXZlcnlfdnRzX290ZlwiXSJ9&_nc_ht=scontent-cdt1-1.cdninstagram.com&_nc_cat=105&_nc_ohc=LPwY_oIyNEoAX-_t3Y9&edm=AABBvjUBAAAA&vs=17915466490947735_383369874&_nc_vs=HBksFQAYJEdBQVRadzVwTHBRRS1LY0FBS2syVkowR3NacEJia1lMQUFBRhUAAsgBABUAGCRHUDZtWnc3S01QZkdaWUFBQUVyVlJINkI3aE5nYmtZTEFBQUYVAgLIAQAoABgAGwAVAAAm6t7X8KHA4T8VAigCQzMsF0BD%2B6XjU%2FfPGBJkYXNoX2Jhc2VsaW5lXzJfdjERAHXqBwA%3D&_nc_rid=e1c1719d78&ccb=7-4&oe=61C73307&oh=00_AT8ZpgC_Lqy9_9qSvw762csPPSmBTv2R_LzBYNRZwoI9vg&_nc_sid=83d603"
[download] Destination: Video by akse_p19 [CTomrHIFc36].mp4
[download] 100% of 1.78MiB in 00:00
```

### PR DESCRIPTION
**Pull request by MinePlayersPE**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

The embed webpage usually isn't locked behind a login wall (for good reason)
Note that some posts don't have video data on embeds, so the 2nd media check is necessary
(Closes https://github.com/yt-dlp/yt-dlp/issues/2093?)

### CODE REVIEW

File: yt_dlp/extractor/instagram.py [hunk 1]
```diff
@@ -293,7 +293,9 @@ def _real_extract(self, url):
         video_id, url = self._match_valid_url(url).group('id', 'url')
         webpage, urlh = self._download_webpage_handle(url, video_id)
         if 'www.instagram.com/accounts/login' in urlh.geturl():
-            self.raise_login_required('You need to log in to access this content')
+            self.report_warning('Main webpage is locked behind the login page.')
```
Comment (pukkandan) [inline yt_dlp/extractor/instagram.py:296]: Do we lose metadata with this? If so, explain that in the warning. If not, warning isn't needed

**Review by pukkandan — COMMENTED** [review]
@Totorrr Can u check if this solves your issue?

File: yt_dlp/extractor/instagram.py (hunk 1 above)
Comment (MinePlayersPE) [reply yt_dlp/extractor/instagram.py:297]: Seems like the only fields missing are `uploader` and `comments` (and `duration` but imo it was already kinda broken in the first place)

**Comment by Totorrr** [comment]
Hi @pukkandan & @MinePlayersPE and thank you for the work on this MR. 

Unfortunately, it doesn't work for me in the video I used as an example. When I access https://www.instagram.com/p/CTomrHIFc36/embed/ with my browser, the page shows "Watch on Instagram" and if I click on that, I'm back to square one (first URL). 

Here is the yt-dlp log: 

```
[debug] Command-line config: ['-v', 'https://www.instagram.com/p/CTomrHIFc36/']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, err UTF-8, pref UTF-8
[debug] yt-dlp version 2021.12.25 [87e049962]
[debug] Python version 3.7.3 (CPython 64bit) - Linux-5.10.0-0.bpo.8-amd64-x86_64-with-debian-10.11
[debug] exe versions: ffmpeg 4.4.1-2 (setts,fdk), ffprobe 4.4.1-2
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [Instagram] Extracting URL: https://www.instagram.com/p/CTomrHIFc36/
[Instagram] CTomrHIFc36: Downloading webpage
WARNING: [Instagram] Main webpage is locked behind the login page. Retrying with embed webpage (Note that some metadata might be missing)
[Instagram] CTomrHIFc36: Downloading embed webpage
ERROR: [Instagram] CTomrHIFc36: You need to log in to access this content. Use --cookies, --username and --password, or --netrc to provide account credentials
  File "/usr/lib/python3/dist-packages/yt_dlp/extractor/common.py", line 599, in extract
    ie_result = self._real_extract(url)
  File "/usr/lib/python3/dist-packages/yt_dlp/extractor/instagram.py", line 324, in _real_extract
    self.raise_login_required('You need to log in to access this content')
  File "/usr/lib/python3/dist-packages/yt_dlp/extractor/common.py", line 1089, in raise_login_required
    raise ExtractorError(msg, expected=True)
```

:(

**Comment by ErraticFox** [comment]
I have tried `youtube-dlp -vU https://www.instagram.com/p/CTomrHIFc36/embed/` six times and they all downloaded with no problem. I've also downloaded another video (my own personal test before I came across this one) at least over 30+ while testing until I came across this fix. I've noticed it only works with `/p/` so if you have a `/reels/` you can replace it with the `/p/` and embed at the end.

**Pull request merged**